test/cpp/client/credentials_test.cc

```diff
@@ -68,7 +91,24 @@ TEST(CredentialsTest, InvalidGoogleRefreshToken) {
 }
 
 TEST(CredentialsTest, DefaultCredentials) {
+  // Simulate a successful detection of metadata server.
+  grpc_core::HttpRequest::SetOverride(mock_successful_metadata_service_response,
```

Comment:
Once you address my comment elsewhere about moving all of the logic into the C-core layer, I think the changes to these tests won't be needed.